### ISSUE TITLE
Warm Cache (Error 500)

### ISSUE THREAD

**Issue by HoloBack** [getgrav/grav-premium-issues#22]
The warm cache plugin suddenly (after updating to 1.7.3) throws an error 500, the error 500 and file permission troubleshooting pages didn't help me out. Any ideas?

<img width="1440" alt="Bildschirmfoto 2021-01-23 um 16 20 21" src="https://user-images.githubusercontent.com/31846205/105605972-f9103a80-5d96-11eb-8127-b8a22f85bfc2.png">

**Comment by rhukster** [comment]
I will take a look at this first chance I get.

**Comment by rhukster** [comment]
Ok, I'm not able to reproduce this on my test setups.  Warm cache is still working fine.  

If you have a look at the "Network" and filter "XHR" then click the "warm cache".  Then when it completes (with 500 error).  Click "warmcache" and look at the response details.  This is mine, but yours should contain more information.

![CleanShot 2021-01-23 at 18 17 04](https://user-images.githubusercontent.com/1084697/105618686-64620880-5da7-11eb-980f-7fa0b21aadc3.gif)

**Comment by rhukster** [comment]
BTW, don't be put off by the `0.1.0` version, that was for testing, this is version `1.0`

**Comment by HoloBack** [comment]
Thanks for your fast replay!
I think I found the issue - the local SSL certificate provided by MAMP is the problem. Is there a workaround for those types of "development certificates"?

{"error":{"type":"Symfony\\Component\\HttpClient\\Exception\\TransportException","message":"SSL peer certificate or SSH remote key was not OK for \"https:\/\/pixit.design.dev:8890\/pixit-design\/sitemap.xml\".","code":0,"file":"\/Users\/Martin\/Sites\/pixit.design.dev\/pixit-design\/vendor\/symfony\/http-client\/Chunk\/ErrorChunk.php","line":64}}

**Comment by rhukster** [comment]
@HoloBack  If you edit the `user/plugins/warm-cache/warm-cache.php` file about line 147:

```php
    public static function warmCache($url = null, $callback = null)
    {
        $config = Grav::instance()['config'];
        $options = ['headers' => ['User-Agent' => $config->get('plugins.warm-cache.user_agent')]];
        $options = array_merge($options, ['verify_peer' => false, 'verify_host' => false]);
```

Add that last line under the `$options = ['headers' => ....` line.

This should add the option to disabling verifying the certificate.. If it works, I'll add this into `warm-cache` for the next release.

Thanks!

**Comment by rhukster** [comment]
I'm so confident this will fix the issue, I just released it as 1.0.1 :)

**Comment by HoloBack** [comment]
Great @rhukster, it works as expected. While testing, I recognized something new - when PHP (I tested with 8.0.1, but version shouldn't matter) is running in CGI-Mode, an internal error 500 occurs after 30s, even if the max_execution_time is set to something much higher (360 in my environment).
